src/main/config/feature.c

```diff
@@ -35,32 +35,74 @@ PG_RESET_TEMPLATE(featureConfig_t, featureConfig,
     .enabledFeatures = DEFAULT_FEATURES | DEFAULT_RX_FEATURE | FEATURE_DYNAMIC_FILTER | FEATURE_ANTI_GRAVITY | FEATURE_AIRMODE,
 );
 
-void featureSet(const uint32_t mask, uint32_t *features)
+static uint32_t runtimeFeatureMask;
+
+void featureInit(void)
+{
+    runtimeFeatureMask = featureConfig()->enabledFeatures;
+}
+
+static void featureSet(const uint32_t mask, uint32_t *features)
 {
     *features |= mask;
 }
 
-void featureClear(const uint32_t mask, uint32_t *features)
+static void featureClear(const uint32_t mask, uint32_t *features)
 {
     *features &= ~(mask);
 }
 
+// Determines if the feature is enabled (active) in the runtime state.
+// This is the primary funciton used by code that wants to know if a
+// feature is available.
 bool featureIsEnabled(const uint32_t mask)
+{
+    return runtimeFeatureMask & mask;
+}
+
+// Determines if the feature is configured (set in the configuration). Doesn't mean the
+// feature is active in the runtime. This function *SHOULD ONLY* be used in the config check
+// performed at startup and when writing to EEPROM.
+bool featureIsConfigured(const uint32_t mask)
 {
     return featureConfig()->enabledFeatures & mask;
 }
 
+// Updates the configuration *AND* runtime state of a feature.
+// Used *ONLY* by the config check process that runs at startup and EEPROM save.
 void featureEnable(const uint32_t mask)
 {
     featureSet(mask, &featureConfigMutable()->enabledFeatures);
+    featureSet(mask, &runtimeFeatureMask);
 }
 
+// Updates the configuration *AND* runtime state of a feature.
+// Used *ONLY* by the config check process that runs at startup and EEPROM save.
 void featureDisable(const uint32_t mask)
+{
+    featureClear(mask, &featureConfigMutable()->enabledFeatures);
+    featureClear(mask, &runtimeFeatureMask);
+}
```

Comment:
I'm almost thinking we want to give these a more distinct name to indicate that they are 'special case' commands that are only good for one specific case. Maybe something like `internalFeatureEnable()` / `internalFeatureDisable()`?

Reply:
I went the other way and made the areas that should only be updating the config be named that way. While I get the "internal" argument, the problem I see is that the `enableFeature()` and `disableFeature()` are also used in a number of target custom `config.c` files. So renaming the functions to `internalEnableFeature()`, etc. would be a bit out of context. Maybe not a big deal since legacy targets and their related `config.c` are a bit deprecated. Instead maybe something like `activateFeature`/`deactivateFeature` might be a middle ground.

Reply:
Yeah, 'internal' isn't good, it was just an illustrative example. 'activate' / 'deactivate' doesn't imply that it works on more than just the feature as it is. How about `configEnableFeature()` / `configDisableFeature()`?